File: common/src/main/java/net/impleri/itemskills/integrations/kubejs/ItemSkillsPlugin.java
package net.impleri.itemskills.integrations.kubejs;

import dev.latvian.mods.kubejs.KubeJSPlugin;
import dev.latvian.mods.kubejs.script.ScriptType;
import net.impleri.itemskills.integrations.kubejs.events.EventsBinding;
import net.impleri.itemskills.integrations.kubejs.events.RestrictionsRegistrationEventJS;
import net.minecraft.server.MinecraftServer;

public class ItemSkillsPlugin extends KubeJSPlugin {
    public static void onStartup(MinecraftServer server) {
        new RestrictionsRegistrationEventJS(server).post(ScriptType.SERVER, EventsBinding.RESTRICTIONS);
    }
}

File: common/src/main/kotlin/net/impleri/itemskills/integrations/kubejs/ItemSkillsPlugin.kt
package net.impleri.itemskills.integrations.kubejs

import dev.architectury.event.events.common.LifecycleEvent
import dev.latvian.mods.kubejs.KubeJSPlugin
import net.impleri.itemskills.integrations.kubejs.events.EventsBinding
import net.impleri.itemskills.integrations.kubejs.events.RestrictionsRegistrationEventJS
import net.minecraft.server.MinecraftServer

class ItemSkillsPlugin : KubeJSPlugin() {
  init {
    LifecycleEvent.SERVER_STARTING.register(
      LifecycleEvent.ServerState { onStartup(it) },
    )
  }

  override fun registerEvents() {
    EventsBinding.GROUP.register()
  }

  companion object {
    fun onStartup(server: MinecraftServer) {
      EventsBinding.RESTRICTIONS.post(RestrictionsRegistrationEventJS(server))
    }
  }
}
